### PR TITLE
YLP-2004 refine dashboard look and feel

### PR DESCRIPTION
**Pull request by fatb38**
## Checklist
- [x] PR name contains the Jira ticket number
- [x] Changelog updated
- [x] Code tested (unit tests and integration tests)
- [x] No console errors/warnings in integration tests
- [x] No translation files updated (except for English files)
- [x] Code formatted on every file
- [x] Copyright added on every file
- [x] End-to-end tests successful on CI

### CODE REVIEW

**Comment by TimCanu** [comment]
Screenshot from 2023-01-20 13-42-17
Maybe the minimum width of this card should be increased not to have part of it being cropped?

File: packages/blip/app/components/chart/index.d.ts [hunk 1]
```diff
@@ -26,82 +26,17 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import MedicalDataService, { Datum } from 'medical-domain'
-import { utils as vizUtils } from 'tidepool-viz'
-import DialogDatePicker from '../../../../yourloops/components/date-pickers/dialog-date-picker'
+import MedicalDataService, { Datum, TimePrefs } from 'medical-domain'
+import DataUtil from 'tidepool-viz/src/utils/data'
```
Comment (TimCanu) [inline packages/blip/app/components/chart/index.d.ts:30]: Weird import... It is usually only `'tidepool-viz'`. Isn't your component exported by viz?

File: packages/blip/app/components/patient-data.js [hunk 2]
```diff
@@ -337,101 +333,109 @@ class PatientDataPage extends React.Component {
...
-            trackMetric={this.trackMetric}
-            updateChartPrefs={this.updateChartPrefs}
-            trendsState={chartStates.trends}
-            onSwitchPatient={this.handleSwitchPatient}
-            onClickNavigationBack={this.handleBackToListButton}
-          />
-        </Route>
-      </Switch>
+      <React.Fragment>
+        <PatientNavBarMemoized
```
Comment (TimCanu) [inline packages/blip/app/components/patient-data.js:344]: Very good! I would suggest to import it as `PatientNavBar`. Most of the components from `dumb` are memoized but exported (from dumb) without the name `Memoized` in them. That way, we know than when looking for the uses of a memoized components, you don't have to search the name with `Memoized` in it.
This is just an idea, maybe I'm wrong so feel free to do whatever you want

File: packages/blip/app/components/patient-data.js [hunk 3]
```diff
@@ -603,6 +607,30 @@ class PatientDataPage extends React.Component {
     this.setState({ createMessageDatetime: null, messageThread: null })
   }
 
+  /**
+   * Handles navigation between patient navbar tabs
+   * @param chart {ChartType} the chart we are going to
+   * @Returns {void}
+   */
+  navigateToChart(chart) {
+    // event.preventDefault()
```
Comment (TimCanu) [inline packages/blip/app/components/patient-data.js:616]: to be removed?

File: packages/dumb/src/components/stats/common/cbg-utils.tsx [hunk 4]
```diff
@@ -50,10 +48,10 @@ export const computeCBGStyle = (value: number, bgClasses: BgClasses): CBGStyle =
     return { backgroundColor: styles['low-background'], color: styles['low-color'], left: '0' }
   }
   if (value > highValue) {
-    return { color: styles['high-color'], backgroundColor: styles['high-background'], left: `${CBG_BAR_WIDTH_IN_PX}px` }
+    return { color: styles['high-color'], backgroundColor: styles['high-background'], left: '100%' }
   }
   const cbgBarRange = highValue - veryLowValue // Number of value included in the cbg bar range (default is from 54 to 250)
-  const left = `${Math.round(((value - veryLowValue) * CBG_BAR_WIDTH_IN_PX) / cbgBarRange)}px`
+  const left = `${Math.round(((value - veryLowValue) * 100) / cbgBarRange)}%`
```
Comment (TimCanu) [inline packages/dumb/src/components/stats/common/cbg-utils.tsx:54]: I suggest keeping a const to store the value `100` because it is unclear what it is (magic number @celia-rouis ) :teacher:

File: packages/tideline/js/tidelinedata.js [hunk 5]
```diff
@@ -207,16 +207,6 @@ function genRandomId() {
   return hexID.join('')
 }
 
-function getTimerFuncs() {
-  // To be able to enable it when needed:
-  if (false) { // eslint-disable-line no-constant-condition
```
Comment (TimCanu) [inline packages/tideline/js/tidelinedata.js:212]: Why did you delete this? It was working

File: packages/tideline/plugins/blip/basics/components/day/HoverDay.js [hunk 6]
```diff
@@ -43,7 +43,14 @@ class HoverDay extends React.Component {
     }
 
     return (
-      <div id={`calendar-day-${type}-${date}-hover`} className={containerClass} onClick={this.handleClickDay} onMouseEnter={this.mouseEnter} onMouseLeave={this.mouseLeave}>
+      <div
+        id={`calendar-day-${type}-${date}-hover`}
+        data-testid={`calendar-day-hover`}
```
Comment (TimCanu) [inline packages/tideline/plugins/blip/basics/components/day/HoverDay.js:48]: What the fuck is this

File: packages/yourloops/components/dashboard-widgets/models/bg-bounds.model.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+export interface BgBounds {
```
Comment (TimCanu) [inline packages/yourloops/components/dashboard-widgets/models/bg-bounds.model.ts:28]: packages/dumb/src/models/blood-glucose.model.ts has the exact same interface in it. Can you use it instead?

File: packages/yourloops/components/dashboard-widgets/models/bg-prefs.model.ts [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { UnitsType } from '../../../lib/units/models/enums/units-type.enum'
+import { BgBounds } from './bg-bounds.model'
+
+export interface BgPrefs {
```
Comment (TimCanu) [inline packages/yourloops/components/dashboard-widgets/models/bg-prefs.model.ts:31]: BGPrefs is also defined in `packages/dumb/src/models/blood-glucose.model.ts`, but it is slightly different... Could you make this work?

File: packages/yourloops/components/dashboard-widgets/patient-dashboard.tsx [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+  const endpoints = [
+    moment.utc(epochDate - msRange).toISOString(), // start
+    moment.utc(epochDate).toISOString() // end
+  ]
+  const showRemoteMonitoringWidget = user.isUserHcp() && getMedicalTeams().some(team => team.monitoring?.enabled)
+  const gridWidgetSize = showRemoteMonitoringWidget ? 4 : 6
+
+  return (
+    <Grid
+      id="patient-dashboard"
```
Comment (TimCanu) [inline packages/yourloops/components/dashboard-widgets/patient-dashboard.tsx:90]: Is this necessary?

File: packages/yourloops/components/dashboard-widgets/patient-dashboard.tsx [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+      paddingX={3}
+    >
+      <Grid
+        item
+        xs={12}
+        display="flex"
+        alignItems="center"
+      >
+        <AccessTime fontSize="small" className="subnav-icon" />
+        <Typography id="subnav-period-label" variant="body1">{t('dashboard-header-period-text')}</Typography>
```
Comment (TimCanu) [inline packages/yourloops/components/dashboard-widgets/patient-dashboard.tsx:104]: is `id="subnav-period-label"` necessary?

File: packages/yourloops/components/dashboard-widgets/patient-dashboard.tsx [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+      </Grid>
+      <Grid item xs={gridWidgetSize}>
+        <PatientStatistics
+          id="dashboard-patient-statistics"
```
Comment (TimCanu) [inline packages/yourloops/components/dashboard-widgets/patient-dashboard.tsx:108]: same, and for all remaining ids in this file

File: packages/yourloops/test/integration/mock/raw-data.mock.ts [hunk 12]
```diff
@@ -0,0 +1,430 @@
+/*
```
Comment (TimCanu) [inline packages/yourloops/test/integration/mock/raw-data.mock.ts:1]: Folder could be:
`packages/yourloops/test/integration/data/data-api.data.ts`

What do you think of this?

File: packages/yourloops/test/integration/pages/patient-dashboard/patient-dashboard-common.spec.tsx [hunk 13]
```diff
@@ -83,10 +84,13 @@ describe('Patient dashboard for anyone', () => {
...
+    expect(await deviceUsageWidget.findByTestId('chart-basics-factory', {}, { timeout: 3000 })).toHaveTextContent('Cartridge changesSunMonTueWedThuFriSatJan 123456789101112131415161718192021')
+    const reservoirChanges = deviceUsageWidget.getAllByTestId('reservoir-change')
+    expect(reservoirChanges).toHaveLength(1)
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/patient-dashboard/patient-dashboard-common.spec.tsx:91]: You seem to have 6 reservoir changes in your data, but only one displayed, how comes? 
Maybe you could clean up your data (remove the useless ones, and gives the human understandable ids that can be useful when debugging)

File: packages/tideline/js/tidelinedata.js (hunk 5 above)
Comment (fatb38) [reply packages/tideline/js/tidelinedata.js:212]: my bad, I thought that a false condition always fails, but maybe it's ok in some other language ?

Comment (TimCanu) [reply packages/tideline/js/tidelinedata.js:212]: Yes you are spot on! Tanguy script would support this kind of thing. :slightly_smiling_face:

**Review by TimCanu — APPROVED** [review]
